sycl/unittests/pi/EventTest.cpp

```diff
@@ -89,7 +89,105 @@ TEST_F(DISABLED_EventTest, PICreateEvent) {
             PI_SUCCESS);
 }
 
-TEST_F(DISABLED_EventTest, piEventGetInfo) {
+struct callback_user_data {
+  pi_int32 event_type;
+  int index;
+};
+
+TEST_F(EventTest, piEventSetCallback) {
+
+  constexpr size_t event_type_count = 3;
+
+  static bool triggered_flag[event_type_count] = {false, false, false};
+
+  pi_int32 event_callback_types[event_type_count] = {
+      PI_EVENT_SUBMITTED, PI_EVENT_RUNNING, PI_EVENT_COMPLETE};
+
+  callback_user_data user_data[event_type_count];
+
+  auto callback = [](pi_event event, pi_int32 status, void *data) {
+    ASSERT_NE(data, nullptr);
+
+    callback_user_data *pdata = static_cast<callback_user_data *>(data);
+
+#ifndef NDEBUG
+    printf("\tEvent callback %d of type %d triggered\n", pdata->index,
+           pdata->event_type);
+#endif
+
+    triggered_flag[pdata->index] = true;
+  };
+
+  // gate event lets us register callbacks before letting the enqueued work be
+  // executed.
+  pi_event gateEvent;
+  ASSERT_EQ((Plugins[0].call_nocheck<detail::PiApiKind::piEventCreate>(
+                _context, &gateEvent)),
+            PI_SUCCESS);
+
+  constexpr const size_t dataCount = 10000u;
+  std::vector<int> data(dataCount);
+  auto size_in_bytes = data.size() * sizeof(int);
+
+  pi_mem memObj;
+  ASSERT_EQ(
+      (Plugins[0].call_nocheck<detail::PiApiKind::piMemBufferCreate>(
+          _context, PI_MEM_FLAGS_ACCESS_RW, size_in_bytes, nullptr, &memObj)),
+      PI_SUCCESS);
+
+  pi_event syncEvent;
+  ASSERT_EQ(
+      (Plugins[0].call_nocheck<detail::PiApiKind::piEnqueueMemBufferWrite>(
+          _queue, memObj, false, 0, size_in_bytes, data.data(), 1, &gateEvent,
+          &syncEvent)),
+      PI_SUCCESS);
+
+  for (size_t i = 0; i < event_type_count; i++) {
+    user_data[i].event_type = event_callback_types[i];
+    user_data[i].index = i;
+    ASSERT_EQ((Plugins[0].call_nocheck<detail::PiApiKind::piEventSetCallback>(
+                  syncEvent, event_callback_types[i], callback, user_data + i)),
+              PI_SUCCESS);
+  }
+
+  ASSERT_EQ((Plugins[0].call_nocheck<detail::PiApiKind::piEventSetStatus>(
+                gateEvent, PI_EVENT_COMPLETE)),
+            PI_SUCCESS);
+  ASSERT_EQ(
+      (Plugins[0].call_nocheck<detail::PiApiKind::piEventsWait>(1, &syncEvent)),
+      PI_SUCCESS);
+
+  bool success = false;
+  // Waiting for callback (max 10 seconds)...
+  for (int i = 0; i < 100; ++i) {
+
+    int sucessful_events = 0;
+
+    std::this_thread::sleep_for(
```

Comment:
Can we just call piQueueFinish instead?

Reply:
I don't see any problem with using `piQueueFinish` instead. Good call.